src/platform/plugins/shared/saved_search/server/saved_objects/schema.ts

```diff
@@ -202,5 +202,31 @@ export const SCHEMA_SEARCH_MODEL_VERSION_9_SO_API_WORKAROUND = schema.object({
   tabs: schema.maybe(tabsV8),
 });
 
-export type DiscoverSessionTabAttributes = TypeOf<typeof DISCOVER_SESSION_TAB_ATTRIBUTES_VERSION_8>;
-export type DiscoverSessionTab = TypeOf<typeof SCHEMA_DISCOVER_SESSION_TAB_VERSION_8>;
+// Version 10 tabs use the same attributes as version 8
+const DISCOVER_SESSION_TAB_ATTRIBUTES_VERSION_10 = DISCOVER_SESSION_TAB_ATTRIBUTES_VERSION_8;
+
+const SCHEMA_DISCOVER_SESSION_TAB_VERSION_10 = SCHEMA_DISCOVER_SESSION_TAB_VERSION_8;
+
+// We need to extend from the base props rather than the full schema to avoid the same error as version 8
+const SCHEMA_SEARCH_MODEL_VERSION_10_EXTENDED = SCHEMA_SEARCH_MODEL_VERSION_8.extends({
+  projectRouting: schema.maybe(schema.nullable(schema.string())),
+  tabs: schema.arrayOf(SCHEMA_DISCOVER_SESSION_TAB_VERSION_10, { minSize: 1 }),
```

Comment:
Do we have to mention `tabs`?

Reply:
We don't have a fix yet for the SO API issue so I think we still need to work around it, but it seems like we could simplify some. Would something like this work instead?
```ts
export const SCHEMA_SEARCH_MODEL_VERSION_10 = SCHEMA_SEARCH_MODEL_VERSION_8.extends({
  projectRouting: schema.maybe(schema.nullable(schema.string())),
});

const { tabs: tabsV10, ...restV10Props } = SCHEMA_SEARCH_MODEL_VERSION_10.getPropSchemas();

export const SCHEMA_SEARCH_MODEL_VERSION_10_SO_API_WORKAROUND = schema.object({
  ...restV10Props,
  tabs: schema.maybe(tabsV10),
});
```

I don't think it's necessary to change `DiscoverSessionTabAttributes` or `DiscoverSessionTab` either.